modules/check_in/spec/services/check_in/vaos/facility_service_spec.rb

```diff
@@ -141,4 +147,94 @@
       end
     end
   end
+
+  describe 'facilities api data from cache' do
+    let(:facility_response) do
+      {
+        id: '500',
+        facilitiesApiId: 'vha_500',
+        vistaSite: '500',
+        vastParent: '500',
+        type: 'va_health_facility',
+        name: 'Johnson & Johnson',
+        classification: 'MC',
+        timezone: {
+          timeZoneId: 'America/New_York'
+        },
+        lat: 32.78447,
+        long: -79.95415,
+        phone: {
+          main: '123-456-7890',
+          fax: '456-892-7890',
+          pharmacy: '632-456-6734',
+          afterHours: '642-632-8932'
+        }
+      }
+    end
+
+    context 'when facility data exists in cache' do
+      before do
+        Rails.cache.write(
+          "check_in.vaos_facility_#{facility_id}",
+          facility_response,
+          expires_in: 12.hours
+        )
+      end
+
+      it 'returns facility data from cache' do
+        svc = subject.new
+        response = svc.get_facility_with_cache(facility_id:)
+        expect_any_instance_of(subject).not_to receive(:perform)
+        expect(response).to eq(facility_response)
+      end
+    end
+
+    context 'when clinic data exists in cache' do
+      let(:clinic_response) do
+        {
+          data: {
+            vistaSite: 534,
+            clinicId: clinic_id,
+            serviceName: 'CHS NEUROSURGERY VARMA',
+            friendlyName: 'CHS NEUROSURGERY VARMA',
+            medicalService: 'SURGERY',
+            physicalLocation: '1ST FL SPECIALTY MODULE 2',
+            phoneNumber: '843-577-5011',
+            stationId: '534',
+            institutionId: '534',
+            stationName: 'Ralph H. Johnson Department of Veterans Affairs Medical Center',
+            primaryStopCode: 406,
+            primaryStopCodeName: 'NEUROSURGERY',
+            secondaryStopCodeName: '*Missing*',
+            appointmentLength: 30,
+            variableAppointmentLength: true,
+            patientDirectScheduling: false,
+            patientDisplay: true,
+            institutionName: 'CHARLESTON VAMC',
+            institutionIEN: '534',
+            institutionSID: '97177',
+            timezone: {
+              timeZoneId: 'America/New_York'
+            },
+            futureBookingMaximumDays: 390
```

Comment:
we really don't need these 20 or so lines, all we are doing is validating data is returned from cache rather than calling `perform` method

Reply:
Sure Gaurav, addressed in [c64415b](https://github.com/department-of-veterans-affairs/vets-api/pull/16711/commits/c64415ba8c256a462347fbb839560b4c250b8b9d)

Reply:
I still see some unnecessary json data that doesn't provide any value to the spec. approving for now